Runtime/SimvaPlugin.cs

```diff
@@ -115,36 +124,57 @@ public bool WantsToQuit()
 
         public void Demo()
         {
-            SceneManager.LoadSceneAsync(GamePlayScene);
+            Log("Starting Demo Gameplay");
+            StartGameplay();
         }
 
         public void OnAuthUpdated(OAuth2Token token)
         {
             lastAuth = token;
         }
 
-        public void RunScene(string name)
+        public void RunScene(string sceneName)
         {
-            if (SceneManager.GetActiveScene().name != SimvaScene)
+            var name="";
+            switch (sceneName)
             {
-                SceneManager.LoadSceneAsync(SimvaScene).completed += ev =>
-                {
-                    DoRunScene(name);
-                };
+                case "Simva.Login":
+                    name = EnableLoginDemoButton ? "Simva.Login.Demo" : "Simva.Login";
+                    break;
+                    
+                case "gameplay":
+                    if (!string.IsNullOrEmpty(GamePlayScene))
+                    {
+                        name = GamePlayScene;
+                    }
+                    else
+                    {
+                        throw new Exception("Please provide your GamePlay Scene name.");
+                    }
+                    break;
+
+                default:
+                    name = sceneName;
+                    break;
             }
-            else
+            if (SceneManager.GetActiveScene().name != name)
             {
                 DoRunScene(name);
             }
         }
 
         private void DoRunScene(string name)
         {
+            Log(name);
```

Comment:
This log should be more descriptive like "Running scene: " + name

Reply:
fixed